src/Serilog/Parameters/PropertyValueConverter.cs

```diff
@@ -154,17 +155,28 @@ LogEventPropertyValue CreatePropertyValue(object value, Destructuring destructur
                 // Only actual dictionaries are supported, as arbitrary types
                 // can implement multiple IDictionary interfaces and thus introduce
                 // multiple different interpretations.
+                var valueAsList = enumerable.Cast<object>();
```

Comment:
I'm curious about this, why would this instruction allocate something? I thought that the `Cast` method would just create a LINQ query that would be later used with the other `Select` query. Is this case `Cast` is not copying the content. Isn't it?

Ref. http://stackoverflow.com/questions/4304961/linq-memory-question